Java/core/src/main/java/com/amazon/randomcutforest/tree/BoxCacheDouble.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.tree;
+
+import java.util.HashMap;
+import java.util.Random;
+
+public class BoxCacheDouble extends BoxCache<double[]> {
```

Comment:
I wonder if we should make the generic `BoxCache<T>` an abstract class, because it looks like it looks like it's not usable without subclassing. We could make `initiateMaps` a protected abstract method.

Reply:
Done, abstract class.